proxy/operator_resource/op_eth_signing_api.py

```diff
@@ -0,0 +1,82 @@
+from .server_abc import OpResourceApi
+from ..base.op_api import OpSignEthMessageRequest, OpSignEthMessageResp, OpSignEthTxRequest, OpSignEthTxResp
+
+
+class OpEthSigningApi(OpResourceApi):
+    name: ClassVar[str] = "OpResource::EthSigning"
+
+    @OpResourceApi.method(name="signEthMessage")
+    async def sign_eth_message(self, request: OpSignEthMessageRequest) -> OpSignEthMessageResp:
+        with logging_context(ctx=request.ctx_id):
+            if not self._cfg.enable_private_api:
+                raise EthError(message="private api is disabled")
+
+            try:
+                data = hex_to_bytes(request.data)
+                message = str.encode(f"\x19Ethereum Signed Message:\n{len(data)}") + data
+                neon_account = await self._neon_account(request.eth_address)
+                signed_message = neon_account.sign_msg(message)
+                return OpSignEthMessageResp(signed_message=signed_message.to_hex())
+            except Exception as exc:
+                _LOG.error("Failed to sign message", exc_info=exc)
+                raise InvalidParamError(message="bad input data")
+
+    @OpResourceApi.method(name="signEthTransaction")
+    async def sign_eth_transaction(self, request: OpSignEthTxRequest) -> OpSignEthTxResp:
+        with logging_context(ctx=request.ctx_id, chain_id=request.chain_id):
+            if not self._cfg.enable_private_api:
+                raise EthError(message="private api is disabled")
+
+            try:
+                if request.tx.from_address.is_empty:
+                    raise InvalidParamError(message="no sender in transaction")
+
+                # We use "data" here because eth-account hasn't been updated
+                # yet, but the correct field is "input"
+                # https://github.com/ethereum/go-ethereum/pull/28078
+
+                tx = {
+                    "to": request.tx.to_address.to_checksum(""),
+                    "chainId": request.chain_id,
+                    "nonce": request.tx.nonce,
+                    "value": request.tx.value,
+                    "data": request.tx.call_data.to_string(""),
+                    "gas": request.tx.gas_limit,
+                    "gasPrice": request.tx.gas_price,
+                }
+
+                neon_account = await self._neon_account(request.tx.from_address)
+                signed_tx = neon_account.sign_transaction(tx)
+
+                signed_neon_tx = request.tx.model_copy(
+                    update={
+                        "neon_tx_hash": EthTxHash.from_raw(signed_tx.hash),
+                        "r": signed_tx.r,
+                        "s": signed_tx.s,
+                        "v": signed_tx.v,
+                    }
+                )
+
+                return OpSignEthTxResp(signed_tx=signed_neon_tx)
+            except Exception as exc:
+                _LOG.error("Failed to sign transaction", exc_info=exc)
+                raise InvalidParamError(message="bad transaction")
```

Comment:
Did you check it? It will not work.
You should return a response type, the result of exceptions is retrying of the request on the client side.